src/anders_mathiesen_ex/ex05/bounded_sim.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Anders Mathiesen'
+__email__ = 'andermat@nmbu.no'
+
+
+from anders_mathiesen_ex.ex05.walker_sim import Walker, Simulation
+
+
+class BoundedWalker(Walker):
+    """
+    Initialise the walker
+
+    Arguments
+    ---------
+    start : int
+        The walker's initial position
+    home : int
+        The walk ends when the walker reaches home
+    left_limit : int
+        The left boundary of walker movement
+    right_limit : int
+        The right boundary  of walker movement
+    """
+    def __init__(self, start, home, left_limit, right_limit):
+        super().__init__(start, home)
+        self.steps_to_go_home = None
+        while left_limit < self.get_position() < right_limit:
+            self.move()
+            if self.is_at_home():
+                self.steps_to_go_home = self.get_position()
+                break
+        self.reset()
```

Comment:
Here you are putting the move-logic in the `__init__` That's definitely not what the code is supposed to do.